GitHub Actions (#1744)
dapr/docs: daprdocs/content/en/developing-applications/integrations/github-actions.md

---
type: docs
weight: 10000

> title: "GitHub Actions"

Reviewer: ```suggestion
title: "Use the Dapr CLI in a GitHub Actions workflow"
```

linkTitle: "GitHub Actions"

> description: "Deploy your application with GitHub Actions"

Reviewer: ```suggestion
description: "Learn how to add the Dapr CLI to your GitHub Actions to deploy and manage Dapr in your environments."
```

---

> Dapr can be integrated with GitHub Actions via the [Dapr tool installer](https://github.com/marketplace/actions/dapr-tool-installer) available in the GitHub Marketplace.

Reviewer: ```suggestion
Dapr can be integrated with GitHub Actions via the [Dapr tool installer](https://github.com/marketplace/actions/dapr-tool-installer) available in the GitHub Marketplace. This installer adds the Dapr CLI to your workflow, allowing you to deploy, manage, and upgrade Dapr across your environments.
```

## Overview

> Before Dapr can be used in your Action Workflow you must install the Dapr CLI on the runner. Using the Dapr tool installer action will install the specified version of the Dapr CLI on macOS, Linux and Windows runners.

Reviewer: ```suggestion
The `dapr/setup-dapr` action will install the specified version of the Dapr CLI on macOS, Linux and Windows runners. Once installed, you can run any [Dapr CLI command]({{< ref cli >}}) to manage your Dapr environments.
```

Reviewer: Also add in the example you have from https://github.com/marketplace/actions/dapr-tool-installer to show the syntax with optional parameters.

## Example

```yaml
- name: Install Dapr

>   uses: dapr/setup-dapr@v1

Reviewer: ```suggestion
  uses: dapr/setup-dapr@v1
  with:
    version: '1.3.0'
```
